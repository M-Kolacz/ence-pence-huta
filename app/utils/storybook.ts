import { type ComponentType, type ComponentProps } from 'react'
import { type MY_VIEWPORTS, allModes } from '../../.storybook/modes.ts'

export const disableControls = <Component extends ComponentType<any>>(
	controls: (keyof ComponentProps<Component>)[],
) => {
	const disabledOptions = {}

	controls.forEach(control =>
		Object.assign(disabledOptions, {
			[control]: {
				table: {
					disable: true,
				},
			},
		}),
	)

	return disabledOptions
}

export const setViewport = (device: keyof typeof MY_VIEWPORTS) => ({
	viewport: {
		defaultViewport: device,
	},
	chromatic: {
		modes: {
			[device]: allModes[device],
		},
	},
})
