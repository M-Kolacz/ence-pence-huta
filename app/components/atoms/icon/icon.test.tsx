import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { getStoriesTestCases } from '#app/utils/storybook.ts'
import * as IconStories from './icon.stories.tsx'

const testCases = getStoriesTestCases(IconStories)

describe('Icon component', () => {
	test.each(testCases)(
		'snapshot test of %s story',
		async (_storyName, Story) => {
			if (!Story) throw new Error('Missing story to test!')

			const { container } = render(<Story />)
			expect(container).toMatchSnapshot()
		},
	)

	test.each(testCases)('a11y test of %s story', async (_storyName, Story) => {
		if (!Story) throw new Error('Missing story to test!')

		const { container } = render(<Story />)

		const result = await axe(container)

		expect(result).toHaveNoViolations()
	})
})
