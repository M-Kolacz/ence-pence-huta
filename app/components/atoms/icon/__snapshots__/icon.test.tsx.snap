// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon component > snapshot test of Component story 1`] = `
<div>
  <svg
    class="inline h-6 w-6 self-center"
  >
    <use
      href="/app/components/atoms/icon/icons/sprite.svg#add"
    />
  </svg>
</div>
`;

exports[`Icon component > snapshot test of Icons story 1`] = `
<div>
  <div
    class="flex max-w-12 flex-wrap gap-4"
  >
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#add"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#baby-shoes"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#blanket"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#breakfast"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#bubble-tea"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#cancel"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#cereal"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#diaper"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#document-money"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#dollars"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#email"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#facebook-verified"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#facebook"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#folder"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#image"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#kindergarten"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#menu"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#milk-bottle"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#newspaper"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#offer"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#ok"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#pacifier"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#phone-call"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#piggy-bank"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#salad"
      />
    </svg>
    <svg
      class="inline h-6 w-6 self-center"
    >
      <use
        href="/app/components/atoms/icon/icons/sprite.svg#web"
      />
    </svg>
  </div>
</div>
`;
